### ISSUE TITLE
Tensorflow Version?

### ISSUE THREAD

**Issue by usmcamp0811** [RuiShu/vae-clustering#2]
First just wanted to say that I really enjoyed your blog post on this and look forward to getting it running. 
Been running into what I have assessed to be Tensorflow version related errors. Had some issues with the Variable Scopes throwing errors, saying that 'reuse' was used multiple times, or something along those lines. I apologize for not having the exact error message.  In my attempts to troubleshoot this I upgraded to TF 1.0 and broke things even more and can't get that error again.

**Comment by RuiShu** [comment]
Yep, I wrote this back when I had TF r0.9. Since then, my custom backend (tensorbayes) has been updated to work for TF r0.12 and I'll soon be switching to TF r1.1. Will probably add tensorbayes as PyPI too so version control isn't an issue in the future. 

I'll update this repo accordingly this weekend. Sorry for the delay! Currently busy with ICML rebuttal ;D

**Comment by usmcamp0811** [comment]
Awesome! Thanks for the quick response!

**Comment by JulienSiems** [comment]
I found your repo very helpful as well! But just like @usmcamp0811 I am having trouble to reproduce the results using the current tensorflow (1.2.1) version. I replaced all tensorbayes dependencies with the respective functions copy pasted directly into my script and made it work under 1.2.1 using tensorflow slim for the layers.  

The results I am getting are very different from the ones in the repo:

M2:
    tr_ent,     tr_loss,       t_ent,        t_loss,        t_acc,        epoch
  2.03e-02,  1.11e+02,  2.08e-02,  1.11e+02,  2.94e-01,  1
    tr_ent,     tr_loss,       t_ent,        t_loss,        t_acc,     epoch
  4.61e-03,  1.03e+02,  5.44e-03,  1.03e+02,  3.17e-01,  16

M2 (modified) and real gaussian mixture vae 
always start at accuracy 0.14 and stay there.

Any thoughts from your side would be greatly appreciated!

**Comment by zeroXzero** [comment]
@JulienSiems  Did you manage to fix this issue. I also face a similar issue when I ported this code to tensorflow 1.3 version. M2 works fine, M2 modified works if I try multiple times and gmvae doesn't work at all. I could see that tr_ent goes very low in the initial epoch itself (in 1e-05 range) and the accuracy remains close to 0.12 .  Is this some initialization problem? I am not able to pinpoint the issue. @RuiShu Any clues?

**Comment by zeroXzero** [comment]
Ok fixed it. The issue was with tensorbayes Dense layers (may be initialization). I replaced them with tf.contrib fully connected layers and the models work as expected :).

**Comment by JesperWestell** [comment]
@zeroXzero Did you get it to work? If so, do you have the code available somewhere? I'm also very interested in replicating the results, but it's hard with a limited knowledge in TensorFlow. My results are currently pretty close Rui's, but it's not quite there yet.

**Comment by zengqg** [comment]
@JesperWestell I have modified RuiShu' s code, and I implement the gmvae only with tensorflow.
